### PR TITLE
Close #12440: Refactor VEHICLE_STATUS to use strong enum

### LINKED ISSUE #12440: Refactor VEHICLE_STATUS to use strong enum

### PR DESCRIPTION
**Pull request by ion232**
Closes #12440 - 'Refactor VEHICLE_STATUS to use strong enum'

### CODE REVIEW

**Comment by tupaschoal** [comment]
You'll also need to sync your fork and rebase to solve merge conflicts: https://github.com/OpenRCT2/OpenRCT2/wiki/Rebase-and-Sync-fork-with-OpenRCT2

File: src/openrct2/ride/Vehicle.h [hunk 1]
```diff
@@ -251,7 +251,7 @@ struct Vehicle : SpriteBase
         int16_t SwingSpeed;
         int16_t crash_z;
     };
-    VEHICLE_STATUS status;
+    Vehicle::Status status;
```
Comment (ZehMatt) [inline src/openrct2/ride/Vehicle.h:254]:
```suggestion
    Status status;
```
No need to specify the scope here.

**Comment by ZehMatt** [comment]
The code looks good, thanks for the PR. Given this is your first PR you should add yourself to the contributors.md if you like to.

**Comment by duncanspumpkin** [comment]
I've merged the other branch and this now has a merge conflict could you fix that then it can be merged. I see you added yourself to contributors.md in another pr

**Comment by ion232** [comment]
Using the wiki method to rebase wasn't working to solve the conflicts this time, so I had to merge 'develop' into my branch instead. Hopefully that won't be an issue.

**Comment by ZehMatt** [comment]
We do not have merges in our commit history. What was the problem trying to rebase? The way I do is done like:
```
git fetch upstream develop
git rebase upstream/develop
```
If you had merge conflicts then you need to solve them first, git will let you know if there are conflicts.

**Comment by ion232** [comment]
@ZehMatt Ok I did the rebase successfully. I'm not used to the rebase process, so I didn't realise that the commits on my branch are applied one at a time to the main branch when rebasing, and that there can be conflicts for each patch. I was only seeing the changes of the first patch and thought my local branch changes had been overwritten by the develop branch after all patches, but it was because I hadn't applied the other patches through --continue. The rebase behaviour wasn't obvious in the wiki, so I had to do some research to figure it out. Anyway it should be fine now.

**Comment by duncanspumpkin** [comment]
It's good knowledge to learn. Rebase makes the history much more linear which means bisecting bugs is easier.

**Comment by ZehMatt** [comment]
Looking good now, can be merged as far as I'm concerned.

**Pull request merged**